pkg/networkmanager/interfaces.js

```diff
@@ -1464,6 +1466,25 @@ function add_usage_monitor(iface) {
                            ];
 }
 
+function complete_settings(settings, device) {
+    if (!device) {
+        console.warn("No device to complete settings", JSON.stringify(settings));
+        return;
+    }
+
+    settings.connection.id = device.Interface;
+    settings.connection.uuid = generate_uuid();
+
+    if (device.DeviceType == 'ethernet') {
+        settings.connection.type = '802-3-ethernet';
+        settings.ethernet = { };
```

Comment:
Good point, this looks odd and can be more defensive. 

Right now, `complete_settings` is only called on incomplete settings that are written as literals in the code, and none of them have a "ethernet" member.  `complete_settings` should never be called on settings that we get from NM.